docs/Running-Tachyon-Fault-Tolerant-on-EC2.md

```diff
@@ -4,4 +4,143 @@ title: Running Tachyon with Fault Tolerance on EC2
+
+# Prerequisites
+
+**Install Vagrant and the AWS plugins**
+
+Download [Vagrant](https://www.vagrantup.com/downloads.html)
+
+Install AWS Vagrant plugin:
+
+    vagrant plugin install vagrant-aws
+    vagrant box add dummy https://github.com/mitchellh/vagrant-aws/raw/master/dummy.box
+
+**Install Tachyon**
+
+Download Tachyon to your local machine, and unzip it:
+
+    $ wget https://github.com/amplab/tachyon/releases/download/v{{site.TACHYON_RELEASED_VERSION}}/tachyon-{{site.TACHYON_RELEASED_VERSION}}-bin.tar.gz
+    $ tar xvfz tachyon-{{site.TACHYON_RELEASED_VERSION}}-bin.tar.gz
+
+**Install python library dependencies**
+
+Install [python>=2.7](https://www.python.org/), not python3.
+
+Under `deploy/vagrant` directory in your home directory, run:
+
+    $ sudo bash bin/install.sh
+
+Alternatively, you can manually install [pip](https://pip.pypa.io/en/latest/installing/), and then in `deploy/vagrant` run:
+
+    $ sudo pip install -r pip-req.txt
+
+# Launch a Cluster
+
+To run a Tachyon cluster on EC2, first sign up for an Amazon EC2 account
+on the [Amazon Web Services site](http://aws.amazon.com/). 
+
+Then create [access keys](http://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSGettingStartedGuide/AWSCredentials.html) 
+and set shell environment variables `AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` by:
+
+    $ export AWS_ACCESS_KEY_ID=<your access key>
+    $ export AWS_SECRET_ACCESS_KEY=<your secret access key>
+
+Next generate your EC2 [Key Pairs](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html). Make sure to set the permissions of your private key file that only you can read it:
+
+    $ chmod 400 <your key pair>.pem
+
+In the configuration file `deploy/vagrant/conf/ec2.yml`, set the value of `Keypair` to your keypair name and `Key_Path` to the path to the pem key.
+
+In the configuration file `deploy/vagrant/conf/tachyon.yml`, set the value of `Masters` to the number of TachyonMasters you want.
+
+By default, the Vagrant script creates a [Security Group](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-network-security.html) named *tachyon-vagrant-test* at [Region(**us-east-1**) and Availability Zone(**us-east-1a**)](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/using-regions-availability-zones.html). The security group will be set up automatically in the region with all inbound/outbound network traffic opened. You can change the security group, region and availability zone in `ec2.yml`.
+
+Now you can launch the Tachyon cluster with Hadoop2.4.1 as under filesystem in us-east-1a by running the script under `deploy/vagrant`:
+
+    ./create <number of machines> aws
+
+Note that the `<number of machines>` above should be larger than or equal to `Masters` set in `deploy/vagrant/conf/tachyon.yml`.
```

Comment:
How do the masters and workers get distributed? Is a worker placed on every machine, or only on the machines without a master?